06-Job-Declaration-Protocol.md

```diff
@@ -195,11 +195,20 @@ Only used in Full-Template mode.
 
 A request sent by JDC that proposes a selected set of transactions to JDS.
 
+For future templates, JDC MUST delay sending this message until the corresponding `SetNewPrevHash` has been received.
+
+After optionally requesting missing txs via `ProvideMissingTransactions`, JDS reconstructs the block using the supplied `prev_hash` and transaction data. It SHOULD pass it to a full node for validation.
+
+If there is a tip mismatch, JDS MAY fail with `invalid-prevhash` but it MAY also:
+- if its tip is ahead of `prev_hash`, retry validation against the older tip
+- if `prev_hash` unknown, briefly wait for a new tip and try again
```

Comment:
I would rephrase them. IMO a JDS can:
1. accept everything and call it a day
2. accept only JDS branch:
    * if declared TIP unknow ....
    * if declared TIP in the past, return the adeguate error code (we need to support **automatic action on error codes** for it), the JDC will wait a little bit and if new block arrive ok if not fail.
3. accept every valid block for every branch with enaugh work ....

Reply:
these 3 options should be well  supported by sv2